src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -320,6 +320,106 @@ void QgsExpressionCalculatorLocatorFilter::triggerResult( const QgsLocatorResult
 {
   QApplication::clipboard()->setText( result.userData.toString() );
 }
+
+// SettingsLocatorFilter
+//
+QgsSettingsLocatorFilter::QgsSettingsLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{}
+
+QgsSettingsLocatorFilter *QgsSettingsLocatorFilter::clone() const
+{
+  return new QgsSettingsLocatorFilter();
+}
+
+void QgsSettingsLocatorFilter::fetchResults( const QString &string, const QgsLocatorContext &, QgsFeedback *feedback )
+{
+  QMap<QString, SettingsHandle> filteredSettingHandlesMap;
+  QMap<QString, QString> *optionsPagesMap = QgisApp::instance()->getOptionsPagesMap();
+
+  for ( QString op : optionsPagesMap->keys() )
+  {
+    if ( feedback->isCanceled() )
+      return;
+
+    if ( stringMatches( op, string ) )
+    {
+      SettingsHandle sh;
+      sh.type = "optionpage";
```

Comment:
You may want to enclose the string in `QStringLiteral( "optionpage" )`

Reply:
Sorry, maybe even better `QLatin1string` in this case (see: https://woboq.com/blog/qstringliteral.html ), if I get it right there is an overload for `==` with `QString` so it should be faster.